File: Tutorial4-8LayerImplementation2/src/main/java/com/test/tutorial4_8layerimplementation2/di/ActivityContributorModule.java
package com.test.tutorial4_8layerimplementation2.di;


import com.test.tutorial4_8layerimplementation2.MainActivity;
import com.test.tutorial4_8layerlibrary.di.ActivityScope;
import com.test.tutorial4_8layerlibrary.di.BaseActivityModule;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityContributorModule {

    @ActivityScope
    @ContributesAndroidInjector(modules = {MainActivityModule.class, BaseActivityModule.class})
    abstract MainActivity contributeMainActivity();

}

File: Tutorial6-4ComponentFactoryDagger-Android-AAC/src/main/java/com/smarttoolfactory/tutorial6_4componentfactorydagger_android/di/ActivityContributorModule.kt
package com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di

import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di.module.MainActivityModule
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di.scope.ActivityScope
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.main.MainActivity
import com.smarttoolfactory.tutorial6_4componentfactorydagger_android.settings.SettingsActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityContributorModule {

    @ActivityScope
    @ContributesAndroidInjector(modules = [MainActivityModule::class, MainActivityFragmentContributorModule::class])
    abstract fun contributeMainActivity(): MainActivity

    @ActivityScope
    @ContributesAndroidInjector(modules = [SettingsActivityFragmentContributorModule::class])
    abstract fun contributeSettingsActivity(): SettingsActivity

}